src/util/triplea/mapXmlCreator/MapPropertiesPanel.java

```diff
@@ -0,0 +1,278 @@
+    final JTextField textFieldWaterFilter = new JTextField(MapXmlCreator.waterFilterString);
+    final JLabel labelMapName = new JLabel("Map Name:");
+    final JLabel labelMapNameExample = new JLabel("e.g. 'Revised'");
+    final JLabel labelMapVersion = new JLabel("Map Version:");
+    final JLabel labelMapVersionExample = new JLabel("e.g. '1.2.0.1'");
+    final JLabel labelResourceName = new JLabel("Resource Name:");
+    final JLabel labelResourceNameExample = new JLabel("e.g. 'PUs'");
+    final JLabel labelMapImageFile = new JLabel("Map Image File:");
+    final JButton buttonSelectMapImageFile = new JButton("Browse");
+    final JLabel labelCentersFile = new JLabel("Map Centers File:");
+    final JButton buttonSelectCentersFile = new JButton("Browse");
+    final JLabel labelWaterFilter = new JLabel("Water Territory Filter");
+    final JLabel labelWaterFilterExample = new JLabel("e.g. 'Sea Zone'");
+
+    GridBagLayout gbl_panel_1 = new GridBagLayout();
+    gbl_panel_1.rowHeights = new int[] {32, 32, 32, 32, 32, 32, 32, 0};
+    gbl_panel_1.rowWeights = new double[] {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE};
+    final JPanel stepActionPanel = mapXMLCreator.getStepActionPanel();
+    stepActionPanel.setLayout(gbl_panel_1);
+
+    // Map Name
+    GridBagConstraints gridBadConstLabelMapName = new GridBagConstraints();
+    gridBadConstLabelMapName.insets = new Insets(0, 0, 5, 5);
+    gridBadConstLabelMapName.anchor = GridBagConstraints.NORTHEAST;
+    gridBadConstLabelMapName.gridheight = 32;
+    gridBadConstLabelMapName.gridy = 1;
+    gridBadConstLabelMapName.gridx = 0;
+    stepActionPanel.add(labelMapName, gridBadConstLabelMapName);
+
+    textFieldMapName.addFocusListener(new FocusListener() {
+
+      @Override
+      public void focusLost(FocusEvent e) {
+        MapXmlHelper.putXmlStrings("info_@name", textFieldMapName.getText());
+      }
+
+      @Override
+      public void focusGained(FocusEvent e) {}
+    });
+    textFieldMapName.setMaximumSize(new Dimension(0, 0));
+    textFieldMapName.setColumns(30);
+    GridBagConstraints gridBadConstTextFieldMapName = (GridBagConstraints) gridBadConstLabelMapName.clone();
+    gridBadConstTextFieldMapName.anchor = GridBagConstraints.NORTH;
+    gridBadConstTextFieldMapName.gridx = 1;
+    textFieldMapName.setMaximumSize(textFieldMapName.getPreferredSize());
+    stepActionPanel.add(textFieldMapName, gridBadConstTextFieldMapName);
+
+    GridBagConstraints gridBadConstLabelMapNameExample = (GridBagConstraints) gridBadConstLabelMapName.clone();
+    gridBadConstLabelMapNameExample.anchor = GridBagConstraints.NORTHWEST;
+    gridBadConstLabelMapNameExample.gridx = 2;
+    stepActionPanel.add(labelMapNameExample, gridBadConstLabelMapNameExample);
+
+    // Map Version
+    stepActionPanel.add(labelMapVersion, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 2, 2));
+
+    textFieldMapVersion.addFocusListener(new FocusListener() {
```

Comment:
Could you be more specific? Why should I create a new method for those 5
lines?
I also do not see commonality to other code places except that they also
create FocustListener

Am 11.04.2016 um 00:08 schrieb Dan Van Atta:
>
> In src/util/triplea/mapXmlCreator/MapPropertiesPanel.java
> <https://github.com/triplea-game/triplea/pull/598#discussion_r59142529>:
>
> > +    textFieldMapName.setColumns(30);
> > +    GridBagConstraints gridBadConstTextFieldMapName = (GridBagConstraints) gridBadConstLabelMapName.clone();
> > +    gridBadConstTextFieldMapName.anchor = GridBagConstraints.NORTH;
> > +    gridBadConstTextFieldMapName.gridx = 1;
> > +    textFieldMapName.setMaximumSize(textFieldMapName.getPreferredSize());
> > +    stepActionPanel.add(textFieldMapName, gridBadConstTextFieldMapName);
> > +
> > +    GridBagConstraints gridBadConstLabelMapNameExample = (GridBagConstraints) gridBadConstLabelMapName.clone();
> > +    gridBadConstLabelMapNameExample.anchor = GridBagConstraints.NORTHWEST;
> > +    gridBadConstLabelMapNameExample.gridx = 2;
> > +    stepActionPanel.add(labelMapNameExample, gridBadConstLabelMapNameExample);
> > +
> > +    // Map Version
> > +    stepActionPanel.add(labelMapVersion, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 2, 2));
> > +
> > +    textFieldMapVersion.addFocusListener(new FocusListener() {
>
> Some common code with the focus listener, extract to a method that
> will create a FocusListener, with parameter that accepts the action of
> which strings to put in the MapXmlHelper.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/triplea-game/triplea/pull/598/files/1420ca8d2fed62a1fa0ecccd1c212a5efef61778..8eddebfb8bb8b20baf4e726edae241ea5aeedd3b#r59142529>
>